### PR TITLE
Show a warning modal when user tries to leave the assember hub

### LINKED ISSUE #41494: [CYS] Hitting browser "back" button on the assembler does nothing
### Expected behavior

Hitting browser "back" button on the assembler should display a modal like hitting "< Let's get creative" button.

Image

### Actual behavior

Hitting browser "back" button on the assembler does nothing

### Steps to reproduce

1. Enable cys feature flag
2. Go to CYS task
3. Complete AI wizard
4. Click on browser "back" button

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #41494 .

This PR shows a warning modal when the user tries to leave the assember hub page by clicking the browser's back button.

One notable change is the removal of popstate and pushState event handlers (https://github.com/woocommerce/woocommerce/pull/41596/commits/6c9db0551eab4958f59513b512a3f2a09b9c769c ) from the iframe. They were interrupting the history and I could not make the back button work as expected without removing them. As far as I can see, everything works as expected, but it should be double-checked. @ilyasfoo ✋ Could you confirm this change please? 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

Make sure you are testing this with a Jetpack connected instance of WooCommerce, as the AI text completion endpoint can only be called with a Jetpack connected account.

1. Create a new JN site for this branch with WooCommerce Beta
2. Enable `customize-store` features flag from `Tools -> WCA Test Helper`
3. Go to `WooCommerce -> Home` and click on `Customize your store` task
4. Proceed to the assember hub page.
5. Click around a few items.
6. Click back button from the assember hub main page.
7. The warning modal should render.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Show a warning modal when the user tries to leave the assember hub (CYS).

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/exit-warning-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import { recordEvent } from '@woocommerce/tracks';
+
+/**
+ * Internal dependencies
+ */
+import { customizeStoreStateMachineEvents } from '..';
+import { ADMIN_URL } from '~/utils/admin-settings';
+import { getNewPath } from '@woocommerce/navigation';
+
+export const ExitWarningModal = ( {
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/customize-store/assembler-hub/exit-warning-modal.tsx]: Can we reuse [this component](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/go-back-warning-modal.tsx#L13)?

**Review by chihsuan — APPROVED** [review]
Left a minor comment. Other than that, everything looks good and tested well! 👍

File: plugins/woocommerce-admin/client/customize-store/assembler-hub/exit-warning-modal.tsx (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/customize-store/assembler-hub/exit-warning-modal.tsx]: @chihsuan Thank you for letting me know 👍 I had no idea we had that. 

Updated in https://github.com/woocommerce/woocommerce/pull/41596/commits/235b8b79fb8d696ca57913bcd9f3682f304db4db and https://github.com/woocommerce/woocommerce/pull/41596/commits/5051fbc677f275c7e5f70a4d5c39985037cdc3fe

**Comment by moon0326** [comment]
Hey @ilyasfoo ✋ Could you check on https://github.com/woocommerce/woocommerce/pull/41596/commits/6c9db0551eab4958f59513b512a3f2a09b9c769c commit and make sure I did not remove anything that shouldn't be removed?

**Review by ilyasfoo — APPROVED** [review]
I confirm iframe's behaviour did not break with the changes. Thanks, @moon0326 !

**Pull request merged**